python/code_challenges/tree_intersection.py

```diff
@@ -1,5 +1,18 @@
 from data_structures.binary_tree import BinaryTree
+from data_structures.hashtable import Hashtable
 
 
-def tree_intersection():
-    pass
+def tree_intersection(tree_a, tree_b):
+    common_numbers = []
+
+    ht = Hashtable()
+    for num in tree_a.pre_order():
+        ht.set(num, num)
+
+    for num in tree_b.pre_order():
+        if ht.contains(num):
```

Comment:
I think the reason this line isn't working as expected is that the hashtable is empty at this point. On line 10, I think the set() method has a bug in it.

As far as why line 10 doesn't work, after throwing a few prints into `hashtable.set()`, it's the `bucket.insert()` line that isn't inserting values. I think `bucket = self.buckets[index]` a few lines above coincidentally sets `bucket` to None at first, and the reference to `self.buckets[index]` is lost. Try switching `bucket` to `self.buckets[index]` on line 68 and 70 in hashtable.py.